File: containers/go-postgres/test-project/hello.go
/*-------------------------------------------------------------------------------------------------------------
 * Copyright (c) Microsoft Corporation. All rights reserved.
 * Licensed under the MIT License. See https://go.microsoft.com/fwlink/?linkid=2090316 for license information.
 *-------------------------------------------------------------------------------------------------------------*/

package main

import (
	"database/sql"
	"fmt"
	"os"

	_ "github.com/lib/pq"
)

func checkError(err error) {
	if err != nil {
		panic(err)
	}

}

func main() {
	var host string = os.Getenv("POSTGRES_HOSTNAME")
	var user string = os.Getenv("POSTGRES_USER")
	var password string = os.Getenv("POSTGRES_PASSWORD")
	var dbname string = os.Getenv("POSTGRES_DB")

	psqlconn := fmt.Sprintf("host=%s user=%s password=%s dbname=%s sslmode=disable", host, user, password, dbname)

	// Ready the Database connection
	db, err := sql.Open("postgres", psqlconn)
	checkError(err)

	// close database connection after it is no longer used
	defer db.Close()

	// check db
	err = db.Ping()
	checkError(err)

	fmt.Println("Connected!")

	fmt.Println("Sending Query to Database")
	rows, err := db.Query(`select datname from pg_database limit 1;`)
	checkError(err)

	// close the query when no longer needed
	defer rows.Close()

	for rows.Next() {
		var datname string

		err = rows.Scan(&datname)
		checkError(err)

		fmt.Printf("One database in this cluster is: %s \n", datname)
	}
}

File: containers/python-3-anaconda-postgres/test-project/hello.py
#-------------------------------------------------------------------------------------------------------------
# Copyright (c) Microsoft Corporation. All rights reserved.
# Licensed under the MIT License. See https://go.microsoft.com/fwlink/?linkid=2090316 for license information.
#-------------------------------------------------------------------------------------------------------------

#%%
import os
import matplotlib
import matplotlib.pyplot as plt
import numpy as np
import psycopg2

try:
    pg_user = os.getenv('POSTGRES_USER')
    pg_password = os.getenv('POSTGRES_PASSWORD')
    pg_db = os.getenv('POSTGRES_DB')
    pg_host = os.getenv('POSTGRES_HOST')
    conn = psycopg2.connect("dbname='{pg_db}' user='{pg_user}' host='{pg_host}' password='{pg_password}'".format(
        pg_db=pg_db,
        pg_user=pg_user,
        pg_host=pg_host,
        pg_password=pg_password
    ))
except:
    print("Unable to connect to the database")
    exit()

try:
    with conn.cursor() as cur:
        cur.execute("""SELECT COUNT(1) from pg_database WHERE datname='postgres'""")
        rows = cur.fetchone()

    if rows[0]==1:
        print("DATABASE CONNECTED")
    else:
        print("ERROR FINDING DATABASE")
finally:
    conn.close()

# # Data for plotting
t = np.arange(0.0, 2.0, 0.01)
s = 1 + np.sin(2 * np.pi * t)

fig, ax = plt.subplots()
ax.plot(t, s)

ax.set(xlabel='time (s)', ylabel='voltage (mV)',
       title='About as simple as it gets, folks')
ax.grid()

fig.savefig("plot.png")
plt.show()

print('Open test-project/plot.png to see the result!')
